File: helpdesk-backend/src/main/java/com/alexandre/helpdesk/entity/Ticket.java
package com.alexandre.helpdesk.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;


@Getter
@Setter
@Entity
@Table(name = "tickets")
public class Ticket {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "titulo")
    private String title;

    @Column(name = "descricao")
    private String description;

    @Enumerated(EnumType.STRING)
    private TicketStatus status;

    @Enumerated(EnumType.STRING)
    @Column(name = "prioridade")
    private TicketPriority priority;

    @Column(name = "responsavel")
    private String responsible;

    @Column(name = "created_at", updatable = false, nullable = false)
    private LocalDateTime createdAt;


}

File: helpdesk-front/src/types/Ticket.ts
export interface Ticket {
  id: number;
  title: string;
  description: string;
  status: 'OPEN' | 'IN_PROGRESS' | 'CLOSED';
  priority: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';
  createdAt: string;
  responsible: string;  
}